app/models/report.rb

```diff
@@ -70,11 +72,13 @@ def links
   end
 
   def location
-    "#{reportable.name} #{reportable.class}"
+    "#{reportable.name} #{reportable_type}"
   end
 
   def sector_name
-    reportable&.sector&.name || ''
+    sector_hsh = hierarchy&.find { |geo| geo['parent_type'] == 'Sector' }
+
+    sector_hsh.present? ? sector_hsh['parent_name'] : 'N/A'
```

Comment:
It's really not that critical that we always return a Sector's name. 99.9% of the time, it will be right in the record's hierarchy, and if it's not, we can just return a 'N/A' string.